vendor/arbiter/arbiter.cpp

```diff
@@ -2819,35 +2888,43 @@ bool AZ::get(
     Headers headers(m_config->baseHeaders());
     headers.insert(userHeaders.begin(), userHeaders.end());
 
-    std::unique_ptr<std::size_t> size(
-            m_config->precheck() && !headers.count("Range") ?
-                tryGetSize(rawPath) : nullptr);
-
     const Resource resource(m_config->baseUrl(), rawPath);
-    const ApiV1 ApiV1(
-            "GET",
-            resource,
-            m_config->authFields(),
-            query,
-            headers,
-            emptyVect);
-
     drivers::Http http(m_pool);
-    Response res(
-            http.internalGet(
-                resource.url(),
-                ApiV1.headers(),
-                ApiV1.query(),
-                size ? *size : 0));
 
-    if (res.ok())
+    std::unique_ptr<Response> res;
+
+    if (m_config->hasSasToken())
     {
-        data = res.data();
+        Query q = m_config->sasToken();
+        q.insert(query.begin(), query.end());
+        res.reset(new Response(http.internalGet(resource.url(), headers, q)));
+    }
+    else
+    {
+        const ApiV1 ApiV1(
+                "GET",
+                resource,
+                m_config->authFields(),
+                query,
+                headers,
+                emptyVect);
+
+        res.reset(
+            new Response(
+                http.internalGet(
+                    resource.url(),
+                    ApiV1.headers(),
+                    ApiV1.query())));
+    }
+
+    if (res->ok())
+    {
+        data = res->data();
         return true;
     }
     else
     {
-        std::cout << res.code() << ": " << res.str() << std::endl;
+        std::cout << res->code() << ": " << res->str() << std::endl;
```

Comment:
This seems like it should throw, log, or something other than just random output.